include/ocpp/v201/charge_point.hpp

```diff
@@ -316,6 +337,26 @@ class ChargePointInterface {
     /// \return vector of composite schedules, one for each evse_id including 0.
     virtual std::vector<CompositeSchedule> get_all_composite_schedules(const int32_t duration,
                                                                        const ChargingRateUnitEnum& unit) = 0;
+
+    /// \brief Gets the configured NetworkConnectionProfile based on the given \p configuration_slot . The
+    /// central system uri of the connection options will not contain ws:// or wss:// because this method removes it if
+    /// present. This returns the value from the cached network connection profiles. \param
+    /// network_configuration_priority \return
+    virtual std::optional<NetworkConnectionProfile>
```

Comment:
Am I correct that these functions are purely used to get parts of the cached network profiles? Might it be an idea to just create one function that returns a const reference to an object containing all information?

Reply:
Jep. I guess we can optimize this by caching the configured network priorities based on their configured priority.

Reply:
Since you also configure the profiles and priorities separately via OCPP, I'm keeping this separate as well. I did rework it so it's no longer a vector of strings.